lib/internal/fs/promises.js

```diff
@@ -668,11 +669,17 @@ async function writeFile(path, data, options) {
     data = Buffer.from(data, options.encoding || 'utf8');
   }
 
+  validateAbortSignal(options.signal);
   if (path instanceof FileHandle)
     return writeFileHandle(path, data, options.signal);
 
+  if (options.signal?.aborted) {
+    throw lazyDOMException('The operation was aborted', 'AbortError');
+  }
+
   const fd = await open(path, flag, options.mode);
   if (options.signal?.aborted) {
+    await fd.close();
     throw lazyDOMException('The operation was aborted', 'AbortError');
   }
   return PromisePrototypeFinally(writeFileHandle(fd, data), fd.close);
```

Comment:
What doo you think of this:
```suggestion
  return PromisePrototypeFinally(
    options.signal?.aborted ?
      PromiseReject(lazyDOMException('The operation was aborted', 'AbortError')) :
      writeFileHandle(fd, data),
    fd.close
  );
```

Reply:
I think that the other way you suggested is better (with the try/finally)